kuma/payments/jinja2/payments/index.html

```diff
@@ -14,3 +14,7 @@
 {% block content %}{% endblock %}
 
 {% block footer %}{% endblock %}
+
+{% block js %}
+<script src="https://checkout.stripe.com/checkout.js" defer></script>
```

Comment:
Pardon my ignorance but won't this happen:
```html
<script defer type="text/javascript" src="/static/react.js" charset="utf-8"></script>
<script src="https://checkout.stripe.com/checkout.js" defer></script>
```
If we can just use Jinja and the `js` block in a way so that it's the other way around then there's no chance that the React `useEffect` happens after the checkout SDK is loaded. 

The bestest solution would be to embed the SDK with something like this: https://github.com/stripe/stripe-js#installation but that's bound to bloat the bundle for all other pages :(

Reply:
Yup, I initially looked at a React-based Stripe implementation but yes, bundle size.

Reply:
So `index.html` extends `react_base.html` which has the following blocks:

```
{% block site_js %}
  {{ providers_media_js() }}
  {% javascript 'react-main' %}
  {% javascript 'auth-modal' %}
  {% for script in scripts %}
    {% javascript script %}
  {% endfor %}
{% endblock %}

{% block js %}{% endblock %}
```

We are not overriding `site_js` inside `index.html` but we simply add Stripe as additional JS using the `js` block:

```
{% block js %}
<script src="https://checkout.stripe.com/checkout.js" defer></script>
{% endblock %}
```

What we want, and what I believe you are suggesting, is that the load order ends up being:

```
<script src="https://checkout.stripe.com/checkout.js" defer></script>
<script defer type="text/javascript" src="/static/react.js" charset="utf-8"></script>
```

and not the current:

```
<script defer type="text/javascript" src="/static/react.js" charset="utf-8"></script>
<script src="https://checkout.stripe.com/checkout.js" defer></script>
```

I wonder if we can safely refactor the above to something like this:

```
{% block site_js %}
  {{ providers_media_js() }}
  {% javascript 'auth-modal' %}
  {% for script in scripts %}
    {% javascript script %}
  {% endfor %}
{% endblock %}

{% block js %}{% endblock %}

{% block react_js %}
  {% javascript 'react-main' %}
{% endblock %}
```

Thinking out loudly here. No testing or experimentation has been done yet 😄

Reply:
Another idea: Load the script dynamically inside the SubscriptionForm component, when it gets mounted (i.e. inside of `useEffect(() => { loadStripe() }, []`). Then you can also track whether stripe is already loaded, if the user manages to get to the stripe modal step before it's ready.

Reply:
Yeah, what @Gregoor said. 

By the way, would you could also have done is drop the `defer` here because then you'd get this:
```
<script defer src="/static/react.js"></script>
<script       src="https://checkout.stripe.com/checkout.js"></script>
```
and then, the browser will load https://checkout.stripe.com/checkout.js before it loads `/static/react.js` but that's not guarantee that it `https://checkout.stripe.com/checkout.js` is ready by the time the `useEffect` triggers deep inside `/static/react.js`.